lib/core/src/client/preview/start.js

```diff
@@ -219,5 +219,62 @@ export default function start(render, { decorateStory } = {}) {
     window.__STORYBOOK_ADDONS_CHANNEL__ = channel; // may not be defined
   }
 
-  return { context, clientApi, configApi, forceReRender };
+  let previousExports = {};
+  const loadStories = (req, framework) => () => {
+    req.keys().forEach(filename => {
+      const fileExports = req(filename);
+
+      // An old-style story file
+      if (!(fileExports.default && fileExports.default.title)) {
+        return;
+      }
+
+      const { default: meta, ...examples } = fileExports;
+
+      const kindName = meta.title || meta.displayName;
+
+      if (previousExports[filename]) {
+        if (previousExports[filename] === fileExports) {
+          return;
+        }
+
+        // Otherwise clear this kind
+        storyStore.removeStoryKind(kindName);
+        storyStore.incrementRevision();
+      }
+
+      // We pass true here to avoid the warning about HMR. It's cool clientApi, we got this
+      const kind = clientApi.storiesOf(kindName, true);
+      kind.addParameters({ framework });
+
+      (meta.decorators || []).forEach(decorator => {
+        kind.addDecorator(decorator);
+      });
+      if (meta.parameters) {
+        kind.addParameters(meta.parameters);
+      }
+
+      Object.keys(examples).forEach(key => {
+        const example = examples[key];
+        const { title = example.title || key, parameters } = example;
```

Comment:
What about `decorators`?

Reply:
Currently decorators are implemented as parameters:

```js
export const foo = () => <Foo />
foo.parameters = {
  decorators: [(storyFn) => <wrapper>{storyFn()}</wrapper>]
}
```

This mirrors the `storiesOf` API. Do you want to hoist this up for convenience and/or consistency with the default export?

Reply:
Oh, ok, that's fine then I guess